ons-spark/spark-functions/logistic-regression.md

```diff
@@ -0,0 +1,1660 @@
+
+````{tabs}
+```{code-tab} r R
+# Access the logistic regression stage of the pipeline
+model_details <- ml_stage(fitted_pipeline, 'generalized_linear_regression')
+
+# Get coefficients and summary statistics from model
+summary <- tibble(coefficients = c(model_details$intercept, model_details$coefficients), 
+                  std_errors = model_details$summary$coefficient_standard_errors(), 
+                  p_values = model_details$summary$p_values()) %>%
+  dplyr::mutate(lower_ci = coefficients - (1.96 * std_errors),
+                upper_ci = coefficients + (1.96 * std_errors))
+                
+# View the summary
+summary
+
+```
+```{code-tab} plaintext R output
+# A tibble: 19 × 5
+   coefficients    std_errors p_values      lower_ci     upper_ci
+          <dbl>         <dbl>    <dbl>         <dbl>        <dbl>
+ 1     1.09          0.372    3.45e- 3       0.359        1.82   
+ 2    -0.680         0.221    2.12e- 3      -1.11        -0.246  
+ 3    -0.000707      0.000985 4.73e- 1      -0.00264      0.00122
+ 4    -0.967         0.226    1.85e- 5      -1.41        -0.524  
+ 5     0.647         1.55     6.77e- 1      -2.40         3.69   
+ 6     0.134         0.0572   1.94e- 2       0.0216       0.246  
+ 7    -0.679         0.366    6.34e- 2      -1.40         0.0379 
+ 8   -25.0      356124.       1.00e+ 0 -698028.      697978.     
+ 9   -26.0      356124.       1.00e+ 0 -698029.      697977.     
+10    -0.0626        1.42     9.65e- 1      -2.84         2.72   
+11     0.202         0.147    1.70e- 1      -0.0864       0.490  
+12    -1.43          0.116    0             -1.66        -1.20   
+13    -0.748         0.0677   0             -0.881       -0.616  
+14    -0.277         0.456    5.44e- 1      -1.17         0.617  
+15    -0.905         0.0921   0             -1.09        -0.725  
+16     0.727         1.42     6.10e- 1      -2.07         3.52   
+17    -0.995         0.160    5.08e-10      -1.31        -0.681  
+18     0.423         0.0614   5.18e-12       0.303        0.544  
+19     0.446         0.0959   3.32e- 6       0.258        0.634  
+```
+````
+You can save a pipeline or pipeline model using the `ml_save` command:
+
+````{tabs}
+```{code-tab} r R
+# Save pipeline
+ml_save(
+  rescue_pipeline,
+  "rescue_pipeline",
+  overwrite = TRUE
+)
+
+# Save pipeline model
+ml_save(
+  fitted_pipeline,
+  "rescue_model",
+  overwrite = TRUE
+)
+```
+````
+
+They can then be re-loaded using the `ml_load()` command and the re-loaded model can be used to re-fit new data with the same model by supplying the name of the new spark dataframe you wish to fit.
+
+````{tabs}
+```{code-tab} r R
+# Reload our saved pipeline
+reloaded_pipeline <- ml_load(sc, "rescue_pipeline")
+
+# Re-fit to a subset of rescue data as an example of how pipelines can be re-used
+new_model <- ml_fit(reloaded_pipeline, sample_frac(rescue, 0.1))
+
+```
```

Comment:
Might be worth having an extra cell where we close the spark session (spark.stop() & spark_disconnect(sc)) as a reminder for good practice